lib/streamlit/runtime/state/safe_session_state.py

```diff
@@ -123,3 +124,7 @@ def __repr__(self):
         kv = ((k, self._state[k]) for k in self._state._keys())
         s = ", ".join(f"{k}: {v!r}" for k, v in kv)
         return f"{{{s}}}"
+
+    def get_query_params(self) -> QueryParams:
+        with self._lock:
+            return self._state._query_params
```

Comment:
I don't think this actually does anything.

If two threads call `get_query_params`, one might wait for a bit while the first thread holds the lock, but eventually both threads will hold a reference to `self._state._query_params`, and at that point there's nothing preventing them from both making changes to the object that we want to ensure mutually exclusive access to.

This might be better defined as a contextmanager that yields `self._state._query_params` while holding the lock so that `QueryParamsProxy` has temporary access to `self._state._query_params` only while within a `with` block. (Or maybe the contextmanager should be the `get_query_params` function that lives in `query_params_proxy.py` -- whichever is cleaner)

Also, we'll want to call `self._yield_callback()` within this method like we do for others in `SafeSessionState`.